common/compare.py

```diff
@@ -0,0 +1,111 @@
+""" Module to find similar models/modules in database to a input model/module
+
+Module to display topk similar models/modules in database to an input parsed 
+model/module. The input model/module is not stored in the database, it is 
+fitted to graph2vec with the models/modules present in the database and 
+cosine similarity is computed using embeddings.
+
+CLA to module and Args to functions in module:
+    file_path (str) : Full path to the model/module to be parsed.
+    file_format (str) : String to denote the file format of the input model/module.
+        One of "TFFrozenGraph", "TFSavedModel" or "TFLite".
+    model_name (str) : Name of the input model/module.
+    include_edge_attrs (str) : Case insensitive string to denote whether to 
+        include edge attributes of input edge from _EDGE_ATTRS in feature,
+        if "true" then they are included.
+    include_node_attrs (str) : Case insensitive string to denote whether to 
+        include node attributes from _NODE_ATTRS in feature, if "true" then 
+        they are included.
+    wl_iterations (int) : Depth of subgraph rooted at every node to be 
+        considered for feature building in graph2vec.
+"""
+
+from sklearn import metrics
+import argparse
+import numpy
+
+from common import Storage
+from common import Vectorize
+from TF.src import TFParser
+from TFLite.src import TFLiteParser
+
+def topk_similar_in_db(file_path, file_format, model_name, include_node_attrs,
+                        include_edge_attrs, topk):
+    """ Module to print topk similar models to a file
+
+    Takes an input model/module and a list of model_graphs present in the 
+    database, parses the file and adds its graph to the models graphs and
+    obtains embeddings for the collective list. Then prints the topk models
+    similar to the input model/module from the database models.
+    """
+
+    # Instance and database ID of spanner database which holds the models
+    INSTANCE_ID = 'ml-models-characterization-db'
+    DATABASE_ID = 'models_db'
+
+    # Parsing models from database into Graph objects
+    storage = Storage.Storage(INSTANCE_ID, DATABASE_ID)
+    model_graphs = storage.parse_models()
+
+    # Parsing file_path depending on file_format
+    if file_format == "TFFrozenGraph":
+        parser = TFParser.TFParser()
+        graph = parser.parse_graph(file_path, model_name, None, None, "False", [])
+
+    elif file_format == "TFSavedModel":
+        parser = TFParser.TFParser()
+        graph = parser.parse_graph(file_path, model_name, None, None, "True", [])
+
+    else:
```

Comment:
Here if the file_format is neither "TFFrozenGraph" nor "TFSavedModel", we assume it would be "TFLiteModel". What if user doesn't give any of these three? Can we throw proper error message in such case?

Reply:
Added an error message. Same is also handled by 'choices' field in argparse.